### DISCUSSION TITLE
What chipher, age using for body encryption when using ssh recipient?

### DISCUSSION THREAD

**Discussion by hexchills** [FiloSottile/age#374] [Q&A] [answered]
Sorry for stupid question and my English... But, my file encrypted with chacha20 or using only ssh key elliptic curve??

**Comment by covert-encryption** [comment, marked as answer]
The data is always encrypted using ChaCha20 with Poly1305 authentication. The **key** for ChaCha encryption comes from SSH or other recipient keys/password that may use elliptic curves for key derivation (or more specifically ECDH over X25519 keys).
